docs/SSH_ARCHITECTURE.md

```diff
@@ -0,0 +1,293 @@
+# SSH Proxy Architecture
+
+Complete documentation of the SSH proxy architecture and operation for Git.
+
+### Main Components
+
+```
+┌─────────────┐         ┌──────────────────┐         ┌──────────┐
+│   Client    │ SSH     │    Git Proxy     │  SSH    │  GitHub  │
+│ (Developer) ├────────→│  (Middleware)    ├────────→│ (Remote) │
+└─────────────┘         └──────────────────┘         └──────────┘
+                              ↓
+                        ┌─────────────┐
+                        │  Security   │
+                        │    Chain    │
+                        └─────────────┘
+```
+
+---
+
+## SSH Host Key (Proxy Identity)
+
+The **SSH host key** is the proxy server's cryptographic identity. It identifies the proxy to clients and prevents man-in-the-middle attacks.
+
+**Auto-generated**: On first startup, git-proxy generates an Ed25519 host key stored in `.ssh/host_key` and `.ssh/host_key.pub`.
+
+**Important**: The host key is NOT used for authenticating to GitHub/GitLab. Agent forwarding handles remote authentication using the client's keys.
+
+**First connection warning**:
+
+```
+The authenticity of host '[localhost]:2222' can't be established.
+ED25519 key fingerprint is SHA256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.
+Are you sure you want to continue connecting (yes/no)?
+```
+
+This is normal! If it appears on subsequent connections, it could indicate the proxy was reinstalled or a potential security issue.
+
+---
+
+## Client → Proxy Communication
+
+### Client Setup
+
+**1. Configure Git remote**:
+
+```bash
+# For GitHub
+git remote add origin ssh://git@git-proxy.example.com:2222/github.com/org/repo.git
+
+# For GitLab
+git remote add origin ssh://git@git-proxy.example.com:2222/gitlab.com/org/repo.git
```

Comment:
Chances are the remote is already present - in that case we need to set it with:

```bash
git remote set-url origin <remote-url>
```